hybrid/flow.py

```diff
@@ -587,13 +588,107 @@ def next(self, states):
         return states[min_idx]
 
 
-class Loop(Runnable):
+class TrackMin(Runnable, traits.SISO):
+    """Tracks (and keeps record of) the "best" (or minimal) :class:`State` seen,
+    according to a metric defined with a `key` function.
+
+    Args:
+        key (callable/str, optional, default=None):
+            Best state is judged according to a metric defined with a `key`.
+            `key` can be a `callable` with a signature::
+
+                key :: (State s, Ord k) => s -> k
+
+            or a string holding a key name/path to be extracted from the input
+            state with `operator.attrgetter` method.
+
+            By default, `key == operator.attrgetter('samples.first.energy')`,
+            thus favoring states containing a sample with the minimal energy.
+
+        output (bool, optional, default=False):
+            Update the output state with the relevant aspect of the best state
+            seen so far.
+
+        input_key (str, optional, default='samples')
+            If `output=True`, then this defines the variable/key name in the
+            input State that shall be included in the output state.
+
+        output_key (str, optional, default='best_samples')
+            If `output=True`, then the key under which the `input_key` from the
+            best state seen so far is stored in the output State.
+
```

Comment:
Here you have both "state" and "State" --> this is getting confusing

Reply:
Fixed.